rapidast.py

```diff
@@ -202,6 +209,81 @@ def dump_rapidast_redacted_configs(main_config_file_location: str, destination_d
             sys.exit(2)
 
 
+def deep_traverse_and_replace(d: dict, suffix: str) -> dict:  # pylint: disable=C0103
+    """
+    Recursively traverse a dictionary and replace key-value pairs where the key ends with `suffix`
+    The value is replaced with the corresponding environment variable value if available.
+    """
+    keys_to_replace = [key for key in d if isinstance(key, str) and key.endswith(suffix)]
+
+    for key in keys_to_replace:
+        new_key = key[: -len(suffix)]
+
+        try:
+            env_value = os.environ[d[key]]
+            d[new_key] = env_value
+            del d[key]
+        except KeyError:
+            logging.error(
+                f"""
+                Environment variable '{d[key]}' referenced by key '{key}' could not be found.
+                No configuration replacement will be made for this key. Please check your configuration and environment"
+                """
+            )
+
+    for key, value in d.items():
+        if isinstance(value, dict):
+            deep_traverse_and_replace(value, suffix)
+        elif isinstance(value, list):
+            for i, item in enumerate(value):
+                if isinstance(item, dict):
+                    value[i] = deep_traverse_and_replace(item, suffix)
+
+    return d
+
+
+def validate_config(config: dict, schema_path: Path) -> bool:
+    """
+    Validate a configuration dictionary against a JSON schema file
+    """
+    try:
+        logging.info("Validating configuration")
+        with schema_path.open("r", encoding="utf-8") as file:
+            schema = json.load(file)
+
+        validate(instance=config, schema=schema, format_checker=Draft202012Validator.FORMAT_CHECKER)
+        logging.info("Configuration is valid")
+        return True
+    except ValidationError as e:
+        logging.error(f"Validation error: {e.message}, json_path: {e.json_path}")
+    except SchemaError as e:
+        logging.error(f"Schema error: {e.message}, json_path: {e.json_path}")
+    except json.JSONDecodeError:
+        logging.error(f"Failed to parse JSON schema: {schema_path}")
+    except Exception as e:  # pylint: disable=W0718
+        logging.error(f"Unexpected error: {e}")
+
+    return False
+
+
+def validate_config_schema(config_file) -> bool:
+    config = yaml.safe_load(load_config_file(config_file))
+
+    try:
+        config_version = config["config"]["configVersion"]
+    except KeyError:
+        logging.error("Missing 'configVersion' in configuration")
+        return False
+
+    schema_path = Path(f"config/schemas/{config_version}/rapidast_schema.json")
```

Comment:
there might be cases where RapiDAST is not run from the rapidast directory. I am not sure if we do a `chdir`. It might be better if we enforce an absolute path using the `rapidast.py`'s location, maybe ?

Reply:
Good point. Updated.